Whiskey/Tasks/Invoke-WhiskeyNspCheck.ps1

```diff
@@ -0,0 +1,167 @@
+        }
+        return
+    }
+
+    $npmRegistryUri = $TaskParameter['NpmRegistryUri']
+    if (-not $npmRegistryUri) 
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message 'Property ''NpmRegistryUri'' is mandatory. It should be the URI to the registry from which Node.js packages should be downloaded, e.g.,
+        
+        BuildTasks:
+        - NspCheck:
+            NpmRegistryUri: https://registry.npmjs.org/
+
+        '
+    }
+
+    $workingDirectory = $TaskContext.BuildRoot
+    if ($TaskParameter['WorkingDirectory'])
+    {
+        $workingDirectory = $TaskParameter['WorkingDirectory'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'WorkingDirectory'
+    }
+
+    Push-Location -Path $workingDirectory
+    try
+    {
+        $noColorArg = ''
+        if (($TaskContext.ByBuildServer) -or ($Host.Name -ne 'ConsoleHost'))
+        {
+            $noColorArg = '--no-color'
+        }
+
+        Write-Timing -Message 'Installing Node.js modules.'
+        Invoke-WhiskeyNpmCommand -NpmCommand 'install' -Argument @('--production=false', $noColorArg) -ApplicationRoot $workingDirectory -RegistryUri $npmRegistryUri -ForDeveloper:$TaskContext.ByDeveloper
+        Write-Timing -Message 'COMPLETE'
+
+        if ($LASTEXITCODE -ne 0)
+        {
+            Stop-WhiskeyTask -TaskContext $TaskContext -Message ('NPM command ''npm install'' failed with exit code ''{0}''.' -f $LASTEXITCODE)
+        }
+
+        if ($TaskContext.ShouldInitialize())
+        {
+            Write-Timing -Message 'Initialization complete.'
+            return
+        }
+
+        $nodePath = Install-WhiskeyNodeJs -RegistryUri $npmRegistryUri -ApplicationRoot $workingDirectory -ForDeveloper:$TaskContext.ByDeveloper
+        $globalNodeModulesRoot = Join-Path -Path ($nodePath | Split-Path) -ChildPath 'node_modules'
+        $nspPath = Join-Path -Path $globalNodeModulesRoot -ChildPath 'nsp\bin\nsp'
+
+        $nspInstallCmd = 'install'
+        if (Test-Path -Path $nspPath -PathType Leaf)
+        {
+            $nspInstallCmd = 'update'
+        }
+        $nspInstallArgs = @('nsp@latest', '-g')
+
+        Write-Timing -Message ('Installing NSP.')
+        Invoke-WhiskeyNpmCommand -NpmCommand $nspInstallCmd -Argument $nspInstallArgs -ApplicationRoot $workingDirectory -RegistryUri $npmRegistryUri -ForDeveloper:$TaskContext.ByDeveloper
+        Write-Timing -Message ('COMPLETE')
+
+        if ($LASTEXITCODE -ne 0)
+        {
+            Stop-WhiskeyTask -TaskContext $TaskContext -Message ('NPM command ''npm {0} {1}'' failed with exit code {2}''' -f $nspInstallCmd,($nspInstallArgs -join ' '),$LASTEXITCODE)
+        }
+
+        Write-Timing -Message ('Running NSP security check.')
+
+        $output = Invoke-Command -NoNewScope -ScriptBlock {
+            & $nodePath $nspPath 'check' '--output' 'json' 2>&1 |
+                ForEach-Object { if( $_ -is [Management.Automation.ErrorRecord]) { $_.Exception.Message } else { $_ } } 

```

Comment:
Why are you treating error records this way?

Reply:
I pulled this line over as is from the `Node` task. I assumed there was a reason this was being done so I left it. Should I remove the special error message handling?

Reply:
Yes. Don't capture any of NSP's output. Just let it return to the user.